Formula/rust.rb

```diff
@@ -36,7 +36,6 @@ class Rust < Formula
   depends_on "cmake" => :build
   depends_on "ninja" => :build
   depends_on "python@3.11" => :build
-  depends_on "libssh2"
```

Comment:
I assume this is due to https://github.com/alexcrichton/ssh2-rs/blob/master/libssh2-sys/build.rs#L19-L22, the linked issue actually relating to Homebrew migration of OpenSSL 1.0 --> 1.1.

Given that we are in the middle of OpenSSL 1.1 --> 3.0, may want to avoid trying to force brew's copy.